src/FieldWarning/Assets/Terrain/Scripts/TerrainMap.cs

```diff
@@ -34,67 +36,314 @@ public class TerrainMap
+
+                if (last == temp || lastcnt == 0)
+                {
+                    lastcnt++;
+                }
+                else
+                {
+                    writer.Write(last);
+                    writer.Write(lastcnt);
+
+                    lastcnt = 1;
+                }
+
+                last = temp;
+
+
+            }
+
+            writer.Write(temp);
+            writer.Write(lastcnt);
+            
+
+            writer.Write((int)'\n');
+
+        }
+
+        writer.Close();
+        
+    }
+    
+    /// <summary>
+    /// This unpacks/uncomporesses the binary height map.
+    /// The compression and structure of this height map is simple.
+    /// Every height value is stored as a fload with a corresponding number of times it appears with
+    /// maybe a newline to designate its now a different x coordinate.
+    /// 
+    /// <height><number_of_times_it_occurs_consecutively>["\n"]
+    /// <4bytes><4bytes><4bytes>
+    /// 
+    /// Everything is 4 bytes for simplicity, including the newline.
+    /// 
+    /// </summary>
+    /// <param name="path"></param>
+    public void ReadHeightMap(string path)
+    {
+        //TODO : not much error checking is done in thsi function
+
+        int nEntry = 2 * mapSize + 2 * EXTENSION;
+
+        // read the entire file into memory
+        var file = File.ReadAllBytes(path);
+
+        //var last_notify_msec = 0;
+        int xCoord = 0;
+        int zCoord = 0;
+
+        BinaryReader reader = new BinaryReader(new MemoryStream(file));
+
+        while (reader.BaseStream.Position != reader.BaseStream.Length)
+        {
+
+            // the sampled height of terrain (4 bytes float) or a newline
+            byte[] heightOrNL = reader.ReadBytes(4);
+
+            // check to see if this is height or newline
+            if (BitConverter.ToInt32(heightOrNL, 0) == (int)0x0a)
+            {
+                zCoord = 0;
+                xCoord++;
+            }
+            else
+            {
+
+                // since we already read a byte but we need 4 bytes
+                int numOfValues = reader.ReadInt32();
+
+                // convert this height into a terrain type.. water, forest etc - byte
+                var bType = (byte)(BitConverter.ToSingle(heightOrNL, 0) > waterHeight ? PLAIN : WATER);
+               
+                // this tells us how far to unpack the compression
+                var zEnd = zCoord + numOfValues;
+  
+                // populate the rest of the same type
+                while (zCoord < zEnd)
+                {
+                    map[xCoord, zCoord] = bType;
+                    zCoord++;
+                }
+            }
+
+            if (_loader != null)
+                // this is our loading screen status
+                _loader.PercentDone = ((double)reader.BaseStream.Position / (double)reader.BaseStream.Length) * 100.0;
```

Comment:
"Assets\Terrain\Scripts\TerrainMap.cs(270,25): error CS1061: 'Loading' does not contain a definition for 'PercentDone' and no accessible extension method 'PercentDone' accepting a first argument of type 'Loading' could be found (are you missing a using directive or an assembly reference?)"

Reply:
correct, everything works locally on my end. Will need to checkout my repo and investigate